File: Java/sortedArrayToBST.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode sortedArrayToBST(int[] num) {
        // IMPORTANT: Please reset any member data you declared, as
        // the same Solution instance will be reused for each test case.
        return sortedArrayToBSTHelper(num, 0, num.length - 1);
    }
    
    private TreeNode sortedArrayToBSTHelper(int[] num, int begin, int end)
    {
        if (begin > end)
	    {
		return null;
	    }
        
        int mid = (end - begin) / 2 + begin;
        TreeNode root = new TreeNode(num[mid]);
        root.left = sortedArrayToBSTHelper(num, begin, mid - 1);
        root.right = sortedArrayToBSTHelper(num, mid + 1, end);
        return root;
    }
}

File: C++/sortedArrayToBST.cc
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode *sortedArrayToBST(vector<int> &num) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        return convertToBSTHelper(num, 0, num.size() - 1);
    }
    
    TreeNode *convertToBSTHelper(vector<int> &num, int low, int high)
    {
        
        if (low > high)
        {
            return NULL;
        }
        else
        {
            int mid = (high - low) / 2 + low;
        
            TreeNode *root = new TreeNode(num[mid]);
            root->left = convertToBSTHelper(num, low, mid - 1);
            root->right = convertToBSTHelper(num, mid + 1, high);
        
            return root;
        }
    }
};